### PR TITLE
Fix 62 sitemap loc

### LINKED ISSUE #62: Location is empty in sitemap
See http://blog.hashobject.com/sitemap.xml.

### PR DESCRIPTION
**Pull request by podviaznikov**
Closes #62. Also contains numerous minor fixes and small refactoring for the markdown task.

### CODE REVIEW

File: src/io/perun.clj [hunk 1]
```diff
@@ -63,7 +70,7 @@
                           (boot/fileset-diff @prev-fs)
                           boot/user-files
                           (boot/by-ext ["md" "markdown"])
-                          (map marshall-tmpfile))
```
Comment (martinklepsch) [inline src/io/perun.clj:66]: if `marshall-tmpfile` isn't used anymore I'd suggest to remove it as well.

File: src/io/perun/markdown.clj [hunk 2]
```diff
@@ -85,20 +83,15 @@
...
-            ; it's duplication of what base task is doing now
-            :path (:path file)
-            :filename (.getName file)})))
+  (let [iofile (io/file (:full-path file))
+        file-content (slurp iofile)
+        md-metadata (parse-file-metadata file-content)
+        html (markdown-to-html file-content options)]
+        ; .getName returns only the filename so this should work cross platform
+        (u/info "Processing Markdown: %s\n" (:filename file))
+        (merge md-metadata {:content html} file)))
```
Comment (martinklepsch) [inline src/io/perun/markdown.clj:92]: I think it's best to print messages like "Processing Markdown..." **before** the actual processing so if it fails it's clear which file caused the failure.

File: src/io/perun/markdown.clj [hunk 3]
```diff
@@ -85,20 +83,15 @@
...
-    ; .getName returns only the filename so this should work cross platform
-    (u/info "Processing Markdown: %s\n" (.getName file))
-    (merge (parse-file-metadata file-content)
-           {:content (markdown-to-html file-content options)
-            ; todo: we will need to remove this from task
-            ; it's duplication of what base task is doing now
-            :path (:path file)
-            :filename (.getName file)})))
+  (let [iofile (io/file (:full-path file))
+        file-content (slurp iofile)
```
Comment (martinklepsch) [inline src/io/perun/markdown.clj:87]: how about

``` clj
(-> file :full-path io/file slurp)
```

matter of taste I guess but saves a line :)

File: src/io/perun/rss.clj [hunk 4]
```diff
@@ -15,7 +15,7 @@
 
 (defn generate-rss-str [files options]
   (let [opts    (select-keys options [:title :description :link])
-        items   (rss-definitions files)
+        items   (rss-definitions (remove #(nil? (:name %)) files))
```
Comment (martinklepsch) [inline src/io/perun/rss.clj:18]:
``` clj
;; Might just do it here? 
(filter :name files)
```

File: src/io/perun/markdown.clj (hunk 2 above)
Comment (podviaznikov) [inline src/io/perun/markdown.clj:92]: makes sense

File: src/io/perun/rss.clj (hunk 4 above)
Comment (podviaznikov) [inline src/io/perun/rss.clj:18]: will do

File: src/io/perun/markdown.clj (hunk 3 above)
Comment (podviaznikov) [inline src/io/perun/markdown.clj:87]: I like your variant better too

File: src/io/perun.clj (hunk 1 above)
Comment (podviaznikov) [inline src/io/perun.clj:66]: yes, wanted to remove it and forgot. thanks for pointing out

**Comment by podviaznikov** [comment]
@martinklepsch applied your feedback. Pushed directly to master

**Comment by martinklepsch** [comment]
@podviaznikov awesome :+1: 

Sorry for just jumping in here and giving unsolicited feedback, I hope you don't mind :blush:

**Comment by podviaznikov** [comment]
@martinklepsch actually opposite:) I would love to get feedback from you and @Deraen _ideally_ on every PR. But unfortunately because of timezone differences and just time constraints for everyone it is not always possible. Anyway your feedback (both Clojure tips and suggestions for perun) are welcome any time

**Comment by martinklepsch** [comment]
Great :)

**Pull request merged**